File: GC.cpp
#include<iostream>
using namespace std;

string gccontent()
{
  string name = " ";
  string dna = "";
  int count = 0;
  double maxpercent = 0;
  string maxname = "";
  while (name != "")
  {
    std::getline(std::cin, name);
    if (name != "")
    {
      name = name.substr(1);
      std::getline(std::cin, dna);
      //dna =CCTGCGGAAGATCGGCACTAGAATAGCCAGAACCGTTTCTCTGAGGCTTCCGGCCTTTCCCACTAATAATTCTGAGG;
      for (int i = 0; i<dna.size(); i++)
      {
        if (dna[i] == 'G' || dna[i] == 'C')
        {
          count++;
        }
      }
      double percent = ((double)count/dna.size())*100;
      if (percent > maxpercent)
      {
        maxpercent = percent;
        maxname = name;
      }
      count = 0;
    }
  }
  return maxname + " " + std::to_string(maxpercent);
}

void main ()
{
  string gc = gccontent();
  int found = gc.find(" ");
  string name = gc.substr(0, found);
  double percent = std::stod(gc.substr(found));
  cout << name << endl;
  cout << percent << endl;

}

File: GC.cs
using System;

public class Program
{
  public static void Main()
  {
    string gc_content = gccontent();
    int index = gc_content.IndexOf(' ');
    if (index != -1)
    {
      Console.WriteLine(gc_content(0, index);
      Console.WriteLine(gc_content(index+1);
    }
  }
  public static string gccontent()
  {
    string name = " ";
    string dnasequence;
    float max = 0;
    string maxname = "";
    while (name != "")
    {
      name = Console.ReadLine();
      if (name == "")
      {
        break;
      }
      else
      {
        int count = 0;
        name = name.Substring(1);
        dnasequence = Console.ReadLine();
        for (int i = 0; i<dnasequence.Length; i++)
        {
          if (dnasequence[i] == 'G' || dnasequence[i] == 'C')
          {
            count++;
          }
        }
        float percent = ((float)count/dnasequence.Length)*100;
        if (percent>max)
        {
          max = percent;
          maxname = dnasequence;
        }
      }
    }
    return max.ToString() + " " + maxname;
  }
}
